src/sinks/statsd.rs

```diff
@@ -266,13 +276,16 @@ impl Encoder<Event> for StatsdEncoder {
             }
         };
 
-        let message = encode_namespace(
-            metric.namespace().or(self.default_namespace.as_deref()),
-            '.',
-            buf.join("|"),
-        );
+        let msg: String = match metric.value() {
+            MetricValue::Distribution { .. } => buf.join("\n"),
+            _ => encode_namespace(
+                metric.namespace().or(self.default_namespace.as_deref()),
+                '.',
+                buf.join("|"),
+            ),
+        };
```

Comment:
I think it would be nice to add a TODO comment above this logic, or perhaps in the earlier match, whichever seems more appropriate to you, outlining the discussion with @tobz about how this is a partial solution, the challenges with applying this to sketches, but that it is something that will need to be completed for the feature to be complete.

In the same vein, it seems like this PR should "address" the issue #11661 . It isn't now, just saying that from my understanding, merging this PR shouldn't close that issue.

Reply:
I added this comment